File: src/test/java/org/cloudfoundry/promregator/config/validations/TargetsHaveConsistentAuthenticatorIdTest.java
package org.cloudfoundry.promregator.config.validations;

import java.util.LinkedList;
import java.util.List;

import org.cloudfoundry.promregator.config.PromregatorConfiguration;
import org.cloudfoundry.promregator.config.Target;
import org.cloudfoundry.promregator.config.TargetAuthenticatorConfiguration;
import org.junit.Assert;
import org.junit.jupiter.api.Test;


public class TargetsHaveConsistentAuthenticatorIdTest {

	@Test
	public void testValidateConfigBrokenNotExistingAtAll() {
		Target t = new Target();
		t.setAuthenticatorId("unittest");
		
		List<Target> targets = new LinkedList<>();
		targets.add(t);
		
		PromregatorConfiguration pc = new PromregatorConfiguration();
		pc.setTargets(targets);
		
		TargetsHaveConsistentAuthenticatorId subject = new TargetsHaveConsistentAuthenticatorId();
		String result = subject.validate(pc);
		
		Assert.assertNotNull(result);
	}
	
	@Test
	public void testValidateConfigBrokenWithWrongTAC() {
		Target t = new Target();
		t.setAuthenticatorId("unittest");
		
		List<Target> targets = new LinkedList<>();
		targets.add(t);
		
		
		TargetAuthenticatorConfiguration tac = new TargetAuthenticatorConfiguration();
		tac.setId("unittestOtherIdentifier");
		List<TargetAuthenticatorConfiguration> tacs = new LinkedList<>();
		tacs.add(tac);
		
		PromregatorConfiguration pc = new PromregatorConfiguration();
		pc.setTargets(targets);
		pc.setTargetAuthenticators(tacs);
		
		TargetsHaveConsistentAuthenticatorId subject = new TargetsHaveConsistentAuthenticatorId();
		String result = subject.validate(pc);
		
		Assert.assertNotNull(result);
	}
	
	@Test
	public void testValidateConfigOk() {
		Target t = new Target();
		t.setAuthenticatorId("unittest");
		
		List<Target> targets = new LinkedList<>();
		targets.add(t);
		
		
		TargetAuthenticatorConfiguration tac = new TargetAuthenticatorConfiguration();
		tac.setId("unittest");
		List<TargetAuthenticatorConfiguration> tacs = new LinkedList<>();
		tacs.add(tac);
		
		PromregatorConfiguration pc = new PromregatorConfiguration();
		pc.setTargets(targets);
		pc.setTargetAuthenticators(tacs);
		
		TargetsHaveConsistentAuthenticatorId subject = new TargetsHaveConsistentAuthenticatorId();
		String result = subject.validate(pc);
		
		Assert.assertNull(result);
	}

}

File: src/test/java/org/cloudfoundry/promregator/config/validations/TargetsHaveConsistentAuthenticatorIdTest.kt
package org.cloudfoundry.promregator.config.validations

import java.util.LinkedList
import org.cloudfoundry.promregator.lite.config.PromregatorConfiguration
import org.cloudfoundry.promregator.lite.config.CfTarget
import org.cloudfoundry.promregator.lite.config.TargetAuthenticatorConfiguration
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class TargetsHaveConsistentAuthenticatorIdTest {
    @Test
    fun testValidateConfigBrokenNotExistingAtAll() {
        val t = CfTarget(authenticatorId = "unittest")
        val targets: MutableList<CfTarget> = LinkedList()
        targets.add(t)
        val pc = PromregatorConfiguration(targets)
        val subject = TargetsHaveConsistentAuthenticatorId()
        val result = subject.validate(pc)
        Assertions.assertNotNull(result)
    }

    @Test
    fun testValidateConfigBrokenWithWrongTAC() {
        val t = CfTarget(authenticatorId = "unittest")
        val targets: MutableList<CfTarget> = LinkedList()
        targets.add(t)
        val tac = TargetAuthenticatorConfiguration()
        tac.id = "unittestOtherIdentifier"
        val tacs: MutableList<TargetAuthenticatorConfiguration> = LinkedList()
        tacs.add(tac)
        val pc = PromregatorConfiguration(targets)
        val subject = TargetsHaveConsistentAuthenticatorId()
        val result = subject.validate(pc)
        Assertions.assertNotNull(result)
    }

    @Test
    fun testValidateConfigOk() {
        val t = CfTarget(authenticatorId = "unittest")
        val tac = TargetAuthenticatorConfiguration().apply { id = "unittest" }
        val pc = PromregatorConfiguration(listOf(t), targetAuthenticators = listOf(tac))
        val subject = TargetsHaveConsistentAuthenticatorId()
        val result = subject.validate(pc)
        Assertions.assertNull(result)
    }
}
